server/auth/impersonate/client.go

```diff
@@ -0,0 +1,87 @@
+package impersonate
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	auth "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Client interface {
+	AccessReview(ctx context.Context, namespace string, verb string, resourceGroup string, resourceType string, resourceName string, subresource string) error
+}
+
+type client struct {
+	kubeClient kubernetes.Interface
+	username   string
+}
+
+func NewClient(kubeClient kubernetes.Interface, username string) (Client, error) {
+	return &client{
+		kubeClient: kubeClient,
+		username:   username,
+	}, nil
+}
+
+func (c *client) AccessReview(ctx context.Context, namespace string, verb string, resourceGroup string, resourceType string, resourceName string, subresource string) error {
+	log.WithFields(log.Fields{
+		"Namespace":   namespace,
+		"Verb":        verb,
+		"Group":       resourceGroup,
+		"Resource":    resourceType,
+		"Name":        resourceName,
+		"Subresource": subresource,
+	}).Debug(fmt.Printf("SubjectAccessReview - %s", c.username))
+
+	review, err := c.kubeClient.AuthorizationV1().SubjectAccessReviews().Create(ctx, &auth.SubjectAccessReview{
```

Comment:
I don't expect this will significantly impact users, as it's only for the argo-server?

Where would you like to include the warning, if you want to?

Reply:
something in the documentation would be fine, you could point out that, as a human must be using it, and humans are slower that automated, it is not expect to impact anyone